### PR TITLE
documentation updates 

### PR DESCRIPTION
**Pull request by ksatirli**
Hey @jcudit and others 👋🏼 

This PR removes some inconsistencies with the docs. Feel free to cherry pick as needed! 

(note: the commit messages that say "updates variable" should have been called "updates reference")

### CODE REVIEW

File: website/docs/r/branch_protection.html.markdown [hunk 1]
```diff
@@ -49,7 +48,11 @@ resource "github_branch_protection" "example" {
 
 }
 
-resource "github_user" "example" {
+resource "github_repository" "example" {
```
Comment (ksatirli) [inline website/docs/r/branch_protection.html.markdown:51]: This resource is needed to make the example code work; I've included it as the examples generally include a _full_ rundown of resources.

File: website/docs/r/branch_protection.html.markdown [hunk 2]
```diff
@@ -49,7 +48,11 @@ resource "github_branch_protection" "example" {
...
+  name = "test"
+}
+
+data "github_user" "example" {
```
Comment (ksatirli) [inline website/docs/r/branch_protection.html.markdown:55]: This was a `github_user` resource, but should have been a data source as far as I can tell.

File: website/docs/r/branch_protection_v3.html.markdown [hunk 3]
```diff
@@ -5,7 +5,7 @@ description: |-
   Protects a GitHub branch using the v3 / REST implementation.  The `github_branch_protection` resource has moved to the GraphQL API, while this resource will continue to leverage the REST API
 ---
 
-# github\_branch\_protection
+# github\_branch\_protection\_v3
```
Comment (ksatirli) [inline website/docs/r/branch_protection_v3.html.markdown:8]: This was missing for the in-page title of this resource (the sidebar title is correct)

**Comment by jcudit** [comment]
👋🏾 @ksatirli thanks for getting all these corrections staged.  Any chance you could take a look at the conflicts? I'd like to get this one merged for the next release.

**Comment by ksatirli** [comment]
Hey @jcudit apologies for the delay - I missed this notification somehow.

Merge conflicts have been removed!

File: website/docs/r/branch_protection_v3.html.markdown [hunk 4]
```diff
@@ -17,9 +17,10 @@ This resource allows you to configure branch protection for repositories in your
 
 ```hcl
 # Protect the main branch of the foo repository. Only allow a specific user to merge to the branch.
-resource "github_branch_protection_v3" "example" {
-  repository     = "${github_repository.example.name}"
+resource " github_branch_protection_v3" "example" {
```
Comment (jcudit) [inline website/docs/r/branch_protection_v3.html.markdown:20]:
```suggestion
resource "github_branch_protection_v3" "example" {
```

The added space may lead to friction when copying.

File: website/docs/r/branch_protection_v3.html.markdown [hunk 5]
```diff
@@ -31,8 +32,8 @@ resource "github_branch_protection_v3" "example" {
 # the "ci/travis" context to be passing and only allow the engineers team merge
 # to the branch.
 
-resource "github_branch_protection_v3" "example" {
-  repository     = "${github_repository.example.name}"
+resource " github_branch_protection_v3" "example" {
```
Comment (jcudit) [inline website/docs/r/branch_protection_v3.html.markdown:35]:
```suggestion
resource "github_branch_protection_v3" "example" {
```

Same here.

**Review by jcudit — APPROVED** [review]
Added a couple of suggestions, but looks good overall! Thanks for taking the time to get these details right for future readers ❤️

**Comment by ksatirli** [comment]
I like those suggestions. Thank you for making this better :)

**Review by jcudit — APPROVED** [review]

**Pull request merged**